client/client.go

```diff
@@ -164,3 +167,20 @@ func WithGroup(group *key.Group) Option {
 		return nil
 	}
 }
+
+// Watcher supplies the `Watch` portion of the drand client interface.
+type Watcher interface {
+	Watch(ctx context.Context) <-chan Result
+}
+
+// WatcherCtor creates a Watcher once a group is known.
+type WatcherCtor func(group *key.Group) (Watcher, error)
+
+// WithWatcher specifies a channel that can provide notifications of new
+// randomness bootstrappeed from the group information.
+func WithWatcher(wc WatcherCtor) Option {
+	return func(cfg *clientConfig) error {
+		cfg.watcher = wc
```

Comment:
also todo is to have `makeClient` above create the watcherClient if specified here

Reply:
Yes. This is slated for a separate PR where all the new/updated client components (failover, gossip, aggregator) will be composed together.